tensorflow/python/framework/sparse_tensor_test.py

```diff
@@ -59,6 +59,13 @@ def testPythonConstruction(self):
         self.assertAllEqual(sess_run_value.values, value.values)
         self.assertAllEqual(sess_run_value.dense_shape, value.dense_shape)
 
+  @test_util.run_all_in_graph_and_eager_modes
```

Comment:
I think this decorator was meant to annotate whole classes - I'm not sure it will work. Anyway, if it doesn't, it's always an option to actually create a tf.function and call that instead, something like this:

def testShape(self):
  @def_function.function
  def test_fn():
    tensor = ...
    tensor = ...
    self.assertEqual(tensor.shape.rank, 2)
  test_fn()

Reply:
Ah that works. Thanks!